### ISSUE TITLE
VoLTE/IMS misbehaving on Mediatek Helio G95

### ISSUE THREAD

**Issue by pranav-vaid** [phhusson/treble_experimentations#2266]
Hello everyone,

I am using Redmi Note 10S with AOSP FLOSS secure v411. This is a mediatek SoC, and VoLTE is working fine, but it is misbehaving. I'll give the scenarios

When I install the ims.apk and check all other options in PHH IMS settings and reboot, VoLTE starts working. But after sometime(maybe a reboot/airplane mode on/off), it stops working. It again starts working if I restart the VoLTE from SIM settings.

Also, I tried Video Calling, but it is not working. In the `*#*#4636#*#*` menu, it is showing registered, but I'm not able to make it. And if the other person is trying to make it, they are getting error that my device doesnot supports it.

**Comment by phhusson** [comment]
Yeah video calling will most likely never work, all implementations I've
seen totally break every Treble expectations, and can't be reasonably
upgraded.

WRT airplane issue, that's noted, if i have some time, i'll look at it,
though I'll consider this pretty low priority issue

Le jeu. 7 avr. 2022 à 17:16, Pranav Vaid ***@***.***> a
écrit :

**Comment by pranav-vaid** [comment]
Hello,

Ahh, okay. As far as I've used this ROM, that seems to be the only issues I encounter. Its a treble rom and the amount of improvements and development stuns me. I am using it as a daily driver. Hats off to all the developers and contributors for this amazing work.

And regarding VoLTE, you take as much time as needed.

Thanks,
Pranav.

**Comment by namanagrawal94** [comment]
Hi PHH, 

Did you get time to see this issue? Same is happening with me. I am on Mediatek helio G80. Volte suddenly stops working after sometime and the call gets routed through 2G. After restarting the phone, volte again starts working.

**Comment by pranav-vaid** [comment]
> Hi PHH,
> 
> Did you get time to see this issue? Same is happening with me. I am on Mediatek helio G80. Volte suddenly stops working after sometime and the call gets routed through 2G. After restarting the phone, volte again starts working.

I guess this is mot fixed yet.

**Comment by pranav-vaid** [comment]
@phhusson
@namanagrawal94 
I think i have found a temporary solution to this. 
![Screenshot_20221211-191440_1](https://user-images.githubusercontent.com/54367393/206907250-37ec41b1-4226-4f03-b17b-d64a59733f4f.png)
In both of these app info menus, go to their battery usage settings, and select 'Unrestricted' in the radio buttons. 

I am using this for a day, and it is working. The IMS is registered even after a airplane mode toggle. 

Anyone with mediatek soc can test tgis and let me know if this works. 

Thanks.
